components/button/index.js

```diff
@@ -25,28 +29,60 @@ export function Button( props, ref ) {
 		className,
 		disabled,
 		focus,
+		children,
 		...additionalProps
 	} = props;
 
+	const tag = href !== undefined && ! disabled ? 'a' : 'button';
+	const tagProps = tag === 'a' ? { href, target, rel } : { type: 'button', disabled };
+	const { icon = true } = additionalProps;
+	const isExternalLink = href && ( target && ! includes( [ '_self', '_parent', '_top' ], target ) );
+
 	const classes = classnames( 'components-button', className, {
 		button: ( isPrimary || isLarge || isSmall ),
 		'button-primary': isPrimary,
 		'button-large': isLarge,
 		'button-small': isSmall,
 		'is-toggled': isToggled,
 		'is-busy': isBusy,
+		'external-link': isExternalLink,
+		'components-icon-button': isExternalLink && icon && ( isPrimary || isLarge || isSmall ),
```

Comment:
One component shouldn't inherit the styles of another. In this case, I wonder if `Button` could render an itself as an `IconButton` when the condition applies (which would trickle down to rendering both the icon and another `Button` where maybe we omit the properties which would satisfy the condition to avoid infinite recursion).

Another idea is to bake in `icon` behavior into `Button`.